### DISCUSSION TITLE
Use AWS X-Ray id generator with auto-instrumentation

### DISCUSSION THREAD

**Discussion by vmihailenco** [open-telemetry/opentelemetry-java-instrumentation#8790] [Q&A] [answered]
Is it possible to use AWS X-Ray id generator with OpenTelemetry Java auto-instrumentation?

**Comment by jkwatson** [comment]
moving this question to the instrumentation repository

**Comment by mateuszrzeszutek** [comment]
Hey @vmihailenco ,

Did you mean the [`AwsXrayIdGenerator`](https://github.com/open-telemetry/opentelemetry-java-contrib/blob/main/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/AwsXrayIdGenerator.java) class?
If this is what you are looking for, you can set a custom `IdGenerator` by implementing a custom [extension](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/examples/extension) that will use the [`AutoConfigurationCustomizerProvider`](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure-spi/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/AutoConfigurationCustomizerProvider.java) SPI to modify the `SdkTracerProviderBuilder` via the [`AutoConfigurationCustomizer#addTracerProviderCustomizer()`](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure-spi/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/AutoConfigurationCustomizer.java#L108-L112) method.

**Reply by vmihailenco** [reply]
Yes, I mean AwsXrayIdGenerator. Any chance OpenTelemetry can provide an extension for AwsXrayIdGenerator?

Ideally, it should be yet another environment variable, e.g.:

```shell
export OTEL_TRACES_IDGENERATOR=xray
```

**Reply by mateuszrzeszutek** [reply, marked as answer]
> Ideally, it should be yet another environment variable, e.g.:
> 
> ```shell
> export OTEL_TRACES_IDGENERATOR=xray
> ```

I wouldn't count on a new OTel-wide env var setting; but I think it's quite possible to have a custom `AutoConfigurationCustomizerProvider` that sets the generator depending on some configuration flag (e.g. `otel.aws-xray.trace-id-generator.enabled` or something) in the [aws-xray](https://github.com/open-telemetry/opentelemetry-java-contrib/tree/main/aws-xray) module. Can you create an issue in the contrib repo and ping the aws-xray component owners?

**Reply by vmihailenco** [reply]
Here is the issue: https://github.com/open-telemetry/opentelemetry-java-contrib/issues/934. I am not sure how to ping the aws-xray component owners...
